lib/medtronic600/medtronic600Simulator.js

```diff
@@ -510,13 +382,122 @@ class Medtronic600Simulator {
     });
   }
 
+  applySuspendedBasals() {
```

Comment:
Yes, I agree - and this is *after* my refactoring! The code that handled suspends was originally in `basal()`, and believe it or not, was *much* more complex than what is now in `basal()` and `applySuspendedBasals()`.
The refactoring came about as a result of me adding a *lot* of unit tests for various simulator scenarios (all based on real event data from Caty's pump). I feel confident that I have all scenarios covered by unit tests (the failures below are because of `node` version incompatibilities, not actual unit tests failures - which I still have to fix).
If you feel that the code is still not maintainable, I can have another look into refactoring it further.